src/url-state-machine.js

```diff
@@ -1227,6 +1225,106 @@ function serializeOrigin(tuple) {
   return result;
 }
 
+function strictlySplitByteSequence(buf, cp) {
```

Comment:
Could we move some of this stuff out into a separate file? usm is already a bit too large, and this is definitely beyond just the URL state machine.